packages/stencil-library/src/components/button/button.tsx

```diff
@@ -1,22 +1,18 @@
 import { Component, Prop, h } from "@stencil/core";
-import { Host, HTMLStencilElement } from "@stencil/core/internal";
-import { Element } from '@stencil/core';
-import { Mode } from "../../interface";
-
-
-export type ButtonVariant = "primary" | "secondary" | "link" | "danger"
+import { ButtonVariant, Mode, IconPosition } from "../../interface";
 
 @Component({
     tag: "njwds-button",
 })
 export class Button {
     @Prop() variant: ButtonVariant = "primary";
-    @Prop() mode: Mode = "light"
-
-    @Prop() asChild: boolean = false
-    @Element() private hostElement: HTMLStencilElement;
+    @Prop() mode: Mode = "light";
+    @Prop() icon?: string;
+    @Prop() iconPosition: IconPosition = "leading";
+    @Prop() iconTitle?: string;
 
     private getButtonClassName(): string {
+
         const getVariantClassName = (variant: ButtonVariant): string => {
```

Comment:
[Dust] Should be this be named `getLightModeVariantClassName` to align with the other 2?